tasks/build.py

```diff
@@ -135,6 +135,46 @@ def create_directory(pr, jobs_base_dir, event_info):
     return ym, pr_id, run_dir
 
 
+def run_cmd(cmd, log_msg='', working_dir=None):
+
+    """Run Commands and logs the process
+
+    Args:
+        cmd (string): command to run
+        log_msg (string): purpose of the commant
+        parent_dir (string): location of arch_job_dir
+
+    Returns:
+        tuple of  2 elements containing
+
+        - process_output(string): output of the process
+        - process_exit_code(string): exit code of the process
+    """
+    if working_dir is None:
+        working_dir = os.getcwd()
+
+    if log_msg:
+        log("'%s' by running '%s' in directory '%s'" % (log_msg, cmd, working_dir))
+    else:
+        log("Running '%s' in directory '%s'" % (cmd, working_dir))
+
+    result = subprocess.run(cmd,
+                            cwd=working_dir,
+                            shell=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    process_output = result.stdout.decode("UTF-8")
+    process_error = result.stderr.decode("UTF-8")
+    process_exit_code = result.returncode
```

Comment:
The `process_` part in the variable names is a bit redundant I think, I would change this to:

```python
    stdout = result.stdout.decode("UTF-8")
    stderr = result.stderr.decode("UTF-8")
    exit_code = result.returncode
```

(and then change below accordingly, of course)